src/airline_agent/constants.py

```diff
@@ -20,19 +21,19 @@
 - list_directory — list directory structure to make more informed searches.
 
 ## Tool Use Guidelines:
-- Keep it tight: aim for 1-2 calls per turn (hard cap 4).
-- Answer only from retrieved content.
+- Don't make more tool calls than necessary.
+- Answer primarily based on information from retrieved content unless the question is simply to clarify broadly understood aspects of commercial air travel (such as standard security procedures, boarding processes, or common airline terminology).
 - If a missing detail blocks tool use, ask one short clarifying question. If not blocking, proceed and state your assumption.
 - Don't dump raw tool output—summarize clearly.
 
 ## Response Guidelines:
-- Answer questions based on information you look up in the knowledge base, not based on your own knowledge.
-- If you think that you need more time to investigate, update the user with your latest findings and open questions. You can proceed if the user confirms.
-- Discuss any airline-related topics with the user.
-- If the user asks about anything unrelated to the airline, politely inform them that you can only assist with airline-related inquiries.
+- Prefer to answer questions based on information you look up in the knowledge base, especially when the question concerns Frontier Airlines–specific products, services, policies, or procedures.
+- For general airline knowledge (e.g., common terms, standard processes, and widely known industry roles), you may give a concise explanation using your general knowledge if the knowledge base does not add important Frontier-specific details.
+- When responding to user, never use phrases like "according to the knowledge base", "I couldn't find anything in the knowledge base", etc. When responding to user, treat the retrieved knowledge base content as your own knowledge, not something you are referencing or searching through.
+- **If the user asks something unrelated to Frontier Airlines or air travel, politely refuse and redirect the conversation. Do not attempt to fulfill or improvise unrelated requests.**
```

Comment:
```suggestion
- If the user asks something unrelated to Frontier Airlines or air travel, politely refuse and redirect the conversation. Do not attempt to fulfill unrelated requests.
```

Reply:
Oh I was having a lot of difficulty with the system still answering general queries and this seemed to help, although I agree possibly not the best solution.

Example queries:
- write a poem about bananas
- how many players can play tennis?

IMO if the Agent answers questions like that without cleanlab then it feels like our system prompt is not strong enough baseline to show the benefit of cleanlab.

Reply:
Got it, then seems fine to keep it bolded.  I wouldn't work too hard to make the system prompt significantly more effective for keeping the Agent on track, as long as the system prompt looks production-grade